www/src/components/CustomThemesForm.tsx

```diff
@@ -0,0 +1,190 @@
+  } catch {
+    return false;
+  }
+}
+
+const CustomThemesForm = forwardRef<CustomThemesFormRef, CustomThemesFormProps>(({ initialTheme, onSave }, ref) => {
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+    trigger,
+    getValues,
+  } = useForm<FormData>({
+    defaultValues: {
+      name: initialTheme?.name || '',
+      urls: initialTheme?.urls.length ? initialTheme.urls.map(url => ({ url })) : [{ url: '' }],
+    },
+    mode: 'onTouched',
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'urls',
+  });
+
+  const urlInputRefs = useRef<(HTMLInputElement | null)[]>([]);
+  const prevUrlsLength = useRef(fields.length);
+
+  useEffect(() => {
+    if (fields.length > prevUrlsLength.current) {
+      urlInputRefs.current[fields.length - 1]?.focus();
+    }
+    prevUrlsLength.current = fields.length;
+  }, [fields.length]);
+
+  const onSubmit = (data: FormData) => {
+    onSave({
+      name: data.name.trim(),
+      urls: data.urls.map(u => u.url.trim()).filter(url => url.length > 0),
+    });
+  };
+
+  const submitForm = async () => {
+    const isValid = await trigger();
+    if (isValid) {
+      const data = getValues();
+      onSubmit(data);
+    }
+  };
+
+  useImperativeHandle(ref, () => ({
+    submitForm,
+  }));
+
+  const handleAddUrl = () => {
+    append({ url: '' });
+  };
+
+  const handleRemoveUrl = (index: number) => {
+    remove(index);
+  };
+
+  return (
+    <Form id="customThemesForm" onSubmit={handleSubmit(onSubmit)}>
+      <Form.Text className="mb-3">
+        Add a custom theme name and one or more CSS URLs to apply your own theme. The CSS files should be
+        accessible via URLs and must end with <code>.css</code>.
+      </Form.Text>
+      <Form.Group controlId="customThemeName" isInvalid={!!errors.name}>
+        <Form.Label>Theme Name</Form.Label>
+        <Form.Control
+          size="sm"
+          {...register('name', {
+            required: 'Theme name is required.',
+            validate: (value) => value.trim().length > 0 || 'Theme name is required.',
+          })}
+        />
+        {errors.name && (
+          <Form.Control.Feedback type="invalid">
+            {errors.name.message}
+          </Form.Control.Feedback>
+        )}
+      </Form.Group>
+      {fields.map((field, idx) => (
+        <Form.Group
+          key={field.id}
+          controlId={`customThemeUrl${idx}`}
+          isInvalid={!!errors.urls?.[idx]}
+        >
+          {idx === 0 && (
+            <Form.Label>CSS URL</Form.Label>
+          )}
+          <Stack direction="horizontal" gap={1}>
+            <Controller
+              name={`urls.${idx}.url`}
+              control={control}
+              rules={{
+                required: 'CSS URL is required.',
+                validate: (value) => {
+                  if (!value || value.trim().length === 0) {
+                    return 'CSS URL is required?!?!.';
```

Comment:
[inform, note to self] The `?!?!` here needs to be removed; it was there only for testing.